web/src/message_events.js

```diff
@@ -121,34 +121,30 @@ export function insert_new_messages(messages, sent_by_this_client) {
     // other lists, so we always update this
     message_util.add_new_messages_data(messages, all_messages_data);
 
-    let render_info;
-
-    if (narrow_state.active()) {
-        // We do this NOW even though the home view is not active,
-        // because we want the home view to load fast later.
-        message_util.add_new_messages(messages, message_lists.home);
-
-        if (narrow_state.filter().can_apply_locally()) {
-            render_info = message_util.add_new_messages(messages, message_lists.current);
-        } else {
-            // if we cannot apply locally, we have to wait for this callback to happen to notify
-            maybe_add_narrowed_messages(
-                messages,
-                message_lists.current,
-                message_util.add_new_messages,
-            );
+    for (const list of message_lists.all_rendered_message_lists()) {
+        if (!list.data.filter.can_apply_locally()) {
+            // If we cannot locally calculate whether the new messages
+            // match the message list, we ask the server whether the
+            // new messages match the narrow, and use that to
+            // determine which new messages to add to the current
+            // message list (or display a notification).
+            maybe_add_narrowed_messages(messages, list, message_util.add_new_messages);
+            continue;
         }
-    } else {
-        // we're in the home view, so update its list
-        render_info = message_util.add_new_messages(messages, message_lists.home);
-    }
 
-    if (sent_by_this_client) {
-        const need_user_to_scroll = render_info && render_info.need_user_to_scroll;
+        // Update the message list's rendering for the newly arrived messages.
+        const render_info = message_util.add_new_messages(messages, list);
+
+        // If this is the currently displayed message list, check if
+        // we need to display a notice that the message appeared out of view.
+        //
         // sent_by_this_client will be true if ANY of the messages
         // were sent by this client; notifications.notify_local_mixes
         // will filter out any not sent by us.
-        notifications.notify_local_mixes(messages, need_user_to_scroll);
+        if (list === message_lists.current && sent_by_this_client) {
```

Comment:
I don't think this is quite right 

* it looks like `list === message_lists.current` is already checked for setting `render_info.need_user_to_scroll`.
* you want to still `notify_local_mixes` even if `need_user_to_scroll` is false, since the message could be outside the current narrow (or other notify reasons) and we still want to notify for that

Reply:
OK -- should we instead be doing `const need_user_to_scroll = list === message_lists.current && sent_by_this_client && render_info && render_info.need_user_to_scroll;`?

Reply:
I think the solution is to add the conditional check *inside* notify_local_mixes, and only for the relevant banners.